### DISCUSSION TITLE
How can I automatically paste a copied link from clipboard without any keypress?

### DISCUSSION THREAD

**Discussion by eduard-buta** [AutomaApp/automa#297] [Q&A] [answered]
Hi,

I'm trying to paste a link from my clipboard into an input field automatically. So far, I've thought of two methods of doing this, but couldn't make any work:
1. Using a custom javascript code that pastes what I have in my clipboard. Couldn't make it work, and since I'm not a programmer, don't know what I'm doing wrong.
2. Tried using two Trigger Event blocks. The first one focuses the field so I can type. And the second was supposed to emulate the CTRL+V keypress combo so I can paste my clipboard value. Couldn't make it work in any way I tried.

Please help me out if you've got a quick fix to this. I'm totally lost.

**Comment by Kholid060** [comment, marked as answer]
You can use javascript code to access the copied link from the [clipboard](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/readText).

```js
navigator.clipboard.readText().then((link) => {
  alert(link);
});
```

But to use the above code, you need to allow the targeted website to access the clipboard.

![image](https://user-images.githubusercontent.com/22908993/151695600-442302e2-061d-4fc8-8689-9c0476fa795f.png)


> Tried using two Trigger Event blocks. The first one focuses the field so I can type. And the second was supposed to emulate the CTRL+V keypress combo so I can paste my clipboard value. Couldn't make it work in any way I tried.

The trigger event block only dispatches the selected event to the element, it doesn't execute keyboard shortcuts.

**Reply by eduard-buta** [reply]
Awesome stuff. However, this only shows me an alert containing the last item in my clipboard. I would like to be able to paste that clipboard item into an input field (say Google Search). Is that possible?

**Reply by Kholid060** [reply]
> Awesome stuff. However, this only shows me an alert containing the last item in my clipboard.

Javascript can only read the last item in the clipboard.

>  would like to be able to paste that clipboard item into an input field (say Google Search). Is that possible?

It's possible, after reading the URL from the clipboard use [querySelector](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector) to select the input field that you want to paste the URL on. For example, changing the Google search input field with the text in the clipboard.

```js
navigator.clipboard.readText().then((link) => {
  const inputField = document.querySelector('.gLFyf');
  inputField.value = link;
});
```
